contracts/community-tokens/contracts/BaseToken.sol

```diff
@@ -41,19 +44,35 @@ contract CollectibleV1 is
         uint256 _maxSupply,
         bool _remoteBurnable,
         bool _transferable,
-        string memory _baseTokenURI
+        string memory _baseTokenURI,
+        address _ownerToken,
+        address _masterToken
     ) ERC721(_name, _symbol) {
         maxSupply = _maxSupply;
         remoteBurnable = _remoteBurnable;
         transferable = _transferable;
         baseTokenURI = _baseTokenURI;
+        ownerToken = _ownerToken;
+        masterToken = _masterToken;
+
+        require(ownerToken != address(0x0) || masterToken != address(0x0), "owner or master tokens required");
```

Comment:
We can save some gas here by using custom errors.
I've raised this before but didn't get further feedback on this.
I'll work on getting these things sorted *before* they hit the codebase here.